v1/transactionrequest/transactionrequest.gunk

```diff
@@ -0,0 +1,659 @@
+}
+
+type ChallengeDetails struct {
+	To          BankAccount  `pb:"1" json:"to"`
+	Amount      types.Amount `pb:"2" json:"amount"`
+	Description string       `pb:"3" json:"description"`
+}
+
+type HistoricTransaction struct {
+	From                   BankAccount  `pb:"1" json:"from"`
+	To                     BankAccount  `pb:"2" json:"to"`
+	Amount                 types.Amount `pb:"3" json:"amount"`
+	Description            string       `pb:"4" json:"description"`
+	Posted                 time.Time    `pb:"5" json:"posted"`
+	Completed              time.Time    `pb:"6" json:"completed"`
+	TransactionRequestType string       `pb:"7" json:"transaction_request_type"`
+	ChargePolicy           string       `pb:"8" json:"charge_policy"`
+}
+
+// TransactionRequestService provides transaction request related operations.
+type TransactionRequestService interface {
+	// AnswerTransactionRequestChallenge sends the answer to the challenge for transaction request
+	//
+	// +gunk http.Match{
+	//         Method: "POST",
+	//         Path:   "/v1/banks/{BankID}/accounts/{AccountID}/transaction-request-types/{TransactionRequestType}/transactionrequest/{TransactionRequestID}/challenge",
+	//         Body:   "*",
+	// }
+	// +gunk openapiv2.Operation{
+	//         Tags:        []string{"TransactionRequest"},
+	//         Description: "Answer the transaction request challenge",
+	//         Summary:     "Answer the transaction reqeust challenge",
+	//         Responses: map[string]openapiv2.Response{
+	//                 "201": openapiv2.Response{
+	//                         Description: "Answered the transaction request challenge",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/transactionrequestAnswerTransactionRequestChallengeResponse",
+	//                                 },
+	//                         },
+	//                 },
+	//         },
+	//         Security: []openapiv2.SecurityRequirement{
+	//                 {
+	//                         SecurityRequirement: map[string]openapiv2.SecurityRequirement_SecurityRequirementValue{
+	//                                 "OAuth2": openapiv2.SecurityRequirement_SecurityRequirementValue{
+	//                                         Scope: []string{
+	//                                                 "write",
+	//                                         },
+	//                                 },
+	//                         },
+	//                 },
+	//         },
+	// }
+	AnswerTransactionRequestChallenge(AnswerTransactionRequestChallengeRequest) AnswerTransactionRequestChallengeResponse
+
+	// CreateAccountTransaction creates a new account transaction request
+	//
+	// +gunk http.Match{
+	//         Method: "POST",
+	//         Path:   "/v1/banks/{BankID}/accounts/{AccountID}/transaction-request-types/ACCOUNT/transactionrequest",
```

Comment:
should we use the uppercase for ACCOUNT?

Reply:
It was like that in the API to differentiate the types ACCCOUNT, ACCOUNT_OTP, SEPA, COUNTERPARTY, and FREE_FORM. I think we should leave it that way. @nkristianto